plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2053,6 +2077,36 @@ static auto maybe_make_debug_time_logger() -> std::optional<decltype(make_debug_
    }
 }
 
+void block_only_sync::schedule() {
+   if (!_pending) {
+      // wait one second to see if we can actually get the block from net plugin before we try to resync from block vault
+      _start_sync_timer.expires_from_now(boost::posix_time::seconds(1));
+      _pending = true;
+      _start_sync_timer.async_wait(
+          app().get_priority_queue().wrap(priority::high, [this](const boost::system::error_code& ec) {
```

Comment:
This should capture a weak pointer of some sort (perhaps this needs to be a method of `_impl` so that it can capture the weak ptr there.  There is a race condition between this and shutdown where `this` could be a stale pointer when this job fires.

Reply:
See the remarks here https://www.boost.org/doc/libs/1_70_0/doc/html/boost_asio/reference/basic_deadline_timer/cancel/overload2.html

This is what the destructor calls and if the timer has expired but the job it posts hasn’t been processed then it will not carry an aborted error code

Reply:
Change to follow the usage example of producer_plugin_impl::_timer which is cancel on plugin shutdown so that we don't have to use week pointer.

Reply:
I think that still has the problem.   in appbase::shutdown we will call two tight loops that call `shutdown` on plugins and then erase them from the list.  There is no event queue pump/draining in between though it may be happening on a different thread depending on how we entered shutdown. 

Regardless, its not safe to assume that the call to `cancel` isn't racing a posted expiration/invocation already in the io_context queue.   Observe that other uses of `_timer` in producer plugin are capturing and locking weak pointers.

Reply:
Fix in commit a406b8513355290fa9efca7873dcc36415d10d8d